packages/core/src/utilities/renderToCanvasGPU.ts

```diff
@@ -0,0 +1,120 @@
+import getOrCreateCanvas from '../RenderingEngine/helpers/getOrCreateCanvas';
+import { ViewportType, Events } from '../enums';
+import StackViewport from '../RenderingEngine/StackViewport';
+import { IImage } from '../types';
+import { getRenderingEngine } from '../RenderingEngine/getRenderingEngine';
+import RenderingEngine from '../RenderingEngine';
+
+/**
+ * Renders an cornerstone image to a Canvas. This method will handle creation
+ * of a temporary enabledElement, setting the imageId, and rendering the image via
+ * a StackViewport, copying the canvas drawing to the given canvas Element, and
+ * disabling the created temporary element. SuppressEvents argument is used to
+ * prevent events from firing during the render process (e.g. during a series
+ * of renders to a thumbnail image).
+ *
+ * @example
+ * ```
+ * const canvas = document.getElementById('myCanvas')
+ *
+ * renderToCanvas(canvas, image)
+ * ```
+ * @param canvas - Canvas element to render to
+ * @param image - The image to render
+ * @param modality - [Default = undefined] The modality of the image
+ * @returns - A promise that resolves when the image has been rendered with the imageId
+ */
+export default function renderToCanvasGPU(
+  canvas: HTMLCanvasElement,
+  image: IImage,
+  modality = undefined,
+  renderingEngineId = '_thumbnails'
+): Promise<string> {
+  if (!canvas || !(canvas instanceof HTMLCanvasElement)) {
+    throw new Error('canvas element is required');
+  }
+
+  const imageIdToPrint = image.imageId;
+  const viewportId = `thumbnailViewport-${imageIdToPrint}`;
+  const imageId = image.imageId;
+  const element = document.createElement('div');
+  element.style.width = `${canvas.width}px`;
+  element.style.height = `${canvas.height}px`;
+
+  // Todo: we should be able to use the temporary element without appending
+  // it to the DOM
+  element.style.visibility = 'hidden';
+  document.body.appendChild(element);
```

Comment:
Hmm, simply adding it to the body with visibility hidden might disrupt the layout of the body. Perhaps consider setting `element.style.position='absolute'`. But I could be wrong here.